File: 3396-valid-word/3396-valid-word.java
class Solution {
    public boolean isValid(String word) {
        if(word.length() < 3){
            return false;
        }

        int vowels = 0;
        int consonants = 0;
        String vowelSet = "aeiouAEIOU";

        for(char c : word.toCharArray()) {
            if (Character.isAlphabetic(c)){
                if(vowelSet.indexOf(c) != -1){
                    vowels++;
                }else{
                    consonants++;
                }
            }else if(!Character.isDigit(c)){
                return false;
            }
        }

        return vowels >= 1 && consonants >= 1;
    }
}

File: 3396-valid-word/3396-valid-word.py
class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False 
        
        vowels = 0
        consonants = 0
        vowelSet = "aeiouAEIOU"

        for c in word:
            if c.isalpha():
                if c in vowelSet:
                    vowels += 1
                else:
                    consonants += 1
            elif not c.isdigit():
                return False
            
        return vowels >= 1 and consonants >= 1